File: examples/openresty_python/lua/aespy.lua
local key = "--(i_love_niu)--"
local iv = "--(i_love_niu)--"
local plain = "iloveniu"
local AES = require "pycrypto_aes"

--现在情况只能加密16* 的字符串，为了能加密任意长度字符，需要一个pad
--加密前加入pad 解密后除去pad pad的选择根据实际情况 { 还算凑合

local addpad = function(plain, pad, size)
    local bsize = size or 16
    local pad = pad or "{"
    local n = bsize - string.len(plain)%bsize
    return plain..string.rep(pad, n)
end

local cleanpad = function(plain, pad)
    local pad = pad or "{"
    local res, _  = string.gsub(plain, pad.."*$", '')
    return res
end

local testfunc = function(key, mode, iv) 
	local aes = AES.new(key, mode, iv)
	local cipher = ngx.encode_base64(aes:encrypt(addpad(plain)))
	ngx.say(cipher)

	local enc = ngx.decode_base64(cipher)
	aes = AES.new(key, mode, iv)
	ngx.say(cleanpad(aes:decrypt(enc )))
end

testfunc(key, AES.MODE_CBC, iv)
--testfunc(key, AES.MODE_CFB, iv)
--testfunc(key, AES.MODE_OFB, iv)

--curl 'http://127.0.0.1:8000/api/aespy'
--pycrypot_aes from https://github.com/orangle/lua-pycrypto-aes

File: examples/openresty_python/aespy.py
# -*- coding=utf-8 -*-
# author: orangleliu
# lua python 无缝加密解密

from Crypto.Cipher import AES
import base64

class AESCipher(object):
    def __init__(self, key, mode , iv, pad="{", size=16):
        self.key = key
        self.mode = mode
        self.pad = pad
        self.size = size
        self.iv = iv

    def encrypt(self, raw):
        cipher = AES.new(self.key, self.mode, self.iv)
        return base64.b64encode(cipher.encrypt(self.addpad(raw)))

    def decrypt(self, enc):
        enc = base64.b64decode(enc)
        cipher = AES.new(self.key, self.mode, self.iv)
        return self.cleanpad(cipher.decrypt(enc))

    def addpad(self, raw):
        n = self.size - len(raw)%self.size
        return raw + self.pad*n

    def cleanpad(self, raw):
        return raw.rstrip(self.pad)


if __name__ == '__main__':
    iv = "--(i_love_niu)--"
    key = "--(i_love_niu)--"
    plain = "iloveniu"

    mode = AES.MODE_CBC
    aes = AESCipher(key, mode, iv)
    cipher = aes.encrypt(plain)
    print cipher
    print aes.decrypt(cipher)

    mode = AES.MODE_CFB
    aes = AESCipher(key, mode, iv)
    cipher = aes.encrypt(plain)
    print cipher
    print aes.decrypt(cipher)

    mode = AES.MODE_OFB
    aes = AESCipher(key, mode, iv)
    cipher = aes.encrypt(plain)
    print cipher
    print aes.decrypt(cipher)
